rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -91,30 +92,55 @@ enum MetricValue {
 
 #[derive(Debug, Deserialize)]
 #[serde(tag = "format", rename_all = "lowercase")]
-enum EncodedSeries<T> {
+enum EncodedSeries<T: Decodable> {
     Array { data: Vec<T> },
+    Base64 { data: String },
 }
 
-impl<T> EncodedSeries<T> {
+impl<T: Decodable> EncodedSeries<T> {
     fn into_vec(self) -> Vec<T> {
```

Comment:
Long term I think we should make this fallible and return `Result<Vec<T>, SomeError>`, the decoding has a few possible failures, e.g. a bad encoding (base64/zstd) or an invalid length of the payload (`length % T::size != 0`).

Reply:
Thanks for the suggestion. I added some error handling, do we think this is sufficient to cover the failure cases you outlined?